### DISCUSSION TITLE
how to train a huge image (like the shape (512, 512, z)) with a deep learning model.

### DISCUSSION THREAD

**Discussion by Ismailok** [Project-MONAI/MONAI#4705] [Q&A]
Hello dears,

I am training a deep learning model (Variational autoencoder) with medical images of dimension (512,512,z) with z varying between 107 and 388 in pytorch. I can't choose a ROI because I need the whole image. 
I want to know, how to integrate a patch step in the model to cut the image in several blocks (sub-images) in order to not have a memory problem or if there is another way to do the training without having a memory problem.

**Comment by wyli** [comment]
could try `monai.data.GridPatchDataset`, I've created some demos for generating 2D patches from volumes https://github.com/Project-MONAI/tutorials/blob/main/modules/2d_slices_from_3d_training.ipynb the demo is about generating patch in `(h, w, 1)` and then squeezed to `(h, w)`, but it could also be configured to have a larger z, say (h, w, 5) and without squeezedim.
